src/dotnet-libman/Contracts/HostInteraction.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager.Tools.Contracts
+{
+    internal class HostInteraction : IHostInteractionInternal
+    {
+        public HostInteraction(EnvironmentSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            Logger = settings.Logger ?? throw new ArgumentNullException(nameof(settings.Logger));
+
+            WorkingDirectory = settings.CurrentWorkingDirectory;
+            CacheDirectory = settings.CacheDirectory;
+        }
+
+        public string WorkingDirectory { get; private set; }
+        public string CacheDirectory { get; }
+        public ILogger Logger { get; }
+
+        public async Task<bool> WriteFileAsync(string path, Func<Stream> content, ILibraryInstallationState state, CancellationToken cancellationToken)
+        {
+            var absolutePath = new FileInfo(Path.Combine(WorkingDirectory, path));
+
+            if (absolutePath.Exists)
+                return true;
```

Comment:
It's also a little odd that we return true when we don't write anything to the file because it already exists.  I know this is an existing implementation, but I question whether it's correct...

Reply:
It looks like the return value is being used to determine if the file should be expected in that location or not.